client/my-sites/email/form/mailboxes/test/validators.ts

```diff
@@ -3,79 +3,328 @@
 
-		formFields.mailbox.value = 'user';
-		formFields.alternativeEmail.value = 'user@test.com';
-		formFields.password.value = 'v4l1dPassWord!';
-		formFields.name.value = 'Letero';
+const provideEmailProviderTestData = (
+	provider: EmailProvider,
+	title: string,
+	existingMailboxNames: string[],
+	fieldValueMap: Partial< Record< FormFieldNames, FieldValue > >,
+	expectedFieldErrorMap: Partial< Record< FormFieldNames, NonNullFieldError > >
+) => {
+	return {
+		provider,
+		title: `${ provider }: ${ title }`,
+		existingMailboxNames,
+		fieldValueMap,
+		expectedFieldErrorMap,
+	};
+};
 
-		mb.validate();
+const provideGoogleTestData = (
+	title: string,
+	fieldValueMap: GoogleTestDataType,
+	expectedFieldErrorMap: Partial< Record< GoogleFormFieldNames, NonNullFieldError > > = {},
+	existingMailboxNames: string[] = []
+) =>
+	provideEmailProviderTestData(
+		EmailProvider.Google,
+		title,
+		existingMailboxNames,
+		fieldValueMap,
+		expectedFieldErrorMap
+	);
 
-		expect( formFields.mailbox.error ).toBeNull();
-		expect( formFields.alternativeEmail.error ).toBeNull();
-		expect( formFields.password.error ).toBeNull();
-		expect( formFields.uuid.error ).toBeNull();
-		expect( formFields.domain.error ).toBeNull();
-	} );
+const provideTitanTestData = (
+	title: string,
+	fieldValueMap: TitanTestDataType,
+	expectedFieldErrorMap: Partial< Record< TitanFormFieldNames, NonNullFieldError > > = {},
+	existingMailboxNames: string[] = []
+) =>
+	provideEmailProviderTestData(
+		EmailProvider.Titan,
+		title,
+		existingMailboxNames,
+		fieldValueMap,
+		expectedFieldErrorMap
+	);
 
-	it( 'should fail validating alternative email', () => {
-		const mb = new MailboxForm( EmailProvider.Titan, 'example.com', [] );
-		const formFields: TitanMailboxFormFields = mb.formFields;
+const createTestDataForGoogle = ( overrides: GoogleTestDataType = {} ): GoogleTestDataType => {
+	return {
+		...{
+			[ FIELD_FIRSTNAME ]: 'First',
+			[ FIELD_LASTNAME ]: 'Last',
+			[ FIELD_MAILBOX ]: 'info',
+			[ FIELD_PASSWORD ]: 'some@password',
+		},
+		...overrides,
+	};
+};
 
-		formFields.alternativeEmail.value = 'sample@example.com';
+const createTestDataForTitan = ( overrides: TitanTestDataType = {} ): TitanTestDataType => {
+	return {
+		...{
+			[ FIELD_ALTERNATIVE_EMAIL ]: 'email.000@gmail.com',
+			[ FIELD_NAME ]: 'Name',
+			[ FIELD_MAILBOX ]: 'info',
+			[ FIELD_PASSWORD ]: '--password',
+		},
+		...overrides,
+	};
+};
 
-		mb.validate();
+const finalTestDataForAllCases = [
+	provideGoogleTestData( 'Valid fields', createTestDataForGoogle() ),
+	provideTitanTestData( 'Valid fields', createTestDataForTitan() ),
+	provideGoogleTestData(
+		'Empty value for required fields should fail validation',
+		createTestDataForGoogle( { [ FIELD_PASSWORD ]: null } ),
+		{ [ FIELD_PASSWORD ]: RequiredValidator.getRequiredFieldError() }
+	),
+	provideTitanTestData(
+		'Empty value for required fields should fail validation',
```

Comment:
Shouldn't we define unique validation messages? Otherwise it may be difficult to identify the actual unit test that failed?

```suggestion
		'Empty value for mailbox field should fail validation',
```